contracts/Wallet.sol

```diff
@@ -1,68 +1,150 @@
-import "./IExtendedERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/IERC721Metadata.sol";
+import "@confluxfans/contracts/token/CRC1155/extensions/ICRC1155Metadata.sol";
+import "@confluxfans/contracts/token/CRC1155/extensions/ICRC1155Enumerable.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
 import "./IWallet.sol";
 import "./TokenListManager.sol";
 
 contract Wallet is IWallet {
+
+    using ERC165Checker for address;
+
     TokenListManager public tokenListManager;
 
     constructor(address _tokenListManager) {
         tokenListManager = TokenListManager(_tokenListManager);
     }
 
-    function getPaginatedTokenInfo(address user, uint start, uint limit) 
-    external view override returns (TokenInfo[] memory tokensInfo) {
-        // Get ERC-20 info
-        address[] memory erc20Tokens = tokenListManager.getWhitelistedTokens(TokenType.ERC20, start, limit);
-        // Get ERC-721 token info
-        address[] memory erc721Tokens = tokenListManager.getWhitelistedTokens(TokenType.ERC721, start, limit);
-        // Get ERC-1155 token info
-        address[] memory erc1155Tokens = tokenListManager.getWhitelistedTokens(TokenType.ERC1155, start, limit);
-
-        // Count total of the token
-        uint256 totalTokens = erc20Tokens.length + erc721Tokens.length + erc1155Tokens.length;
-        TokenInfo[] memory paginatedInfo = new TokenInfo[](totalTokens);
-        uint256 count = 0;
-
-        // Fill in ERC-20 token information
-        for (uint256 i = 0; i < erc20Tokens.length; i++) {
-            IExtendedERC20 token = IExtendedERC20(erc20Tokens[i]);
-            paginatedInfo[count++] = TokenInfo({
-                tokenAddress: erc20Tokens[i],
-                balance: token.balanceOf(user),
-                name: token.name(),
-                symbol: token.symbol(),
-                decimals: token.decimals()
-            });
+    // Private helper function to determine the type of token
+    function getTokenType(address tokenAddress) private view returns (TokenType) {
+        // Checking the token's interface via ERC165
+        if (tokenAddress.supportsInterface(type(IERC721).interfaceId)) {
+            return TokenType.ERC721;
+        } else if (tokenAddress.supportsInterface(type(IERC1155).interfaceId)) {
+            return TokenType.ERC1155;
+        } else {
+            return TokenType.ERC20; // Defaults to ERC-20 if not ERC-721 or ERC-1155
         }
+    }
 
-        // Fill in ERC-721 token information
-        for (uint256 i = 0; i < erc721Tokens.length; i++) {
-            paginatedInfo[count++] = TokenInfo({
-                tokenAddress: erc721Tokens[i],
-                balance: 0,
-                name: "",
-                symbol: "",
-                decimals: 0
-            });
+    // Private function to get information about a token
+    function getTokenInfo(address tokenAddress, address user) private view returns (TokenInfo[] memory) {
```

Comment:
Why return asset array for a single token?

Reply:
This is to maintain a consistent array return type across all token types, I could try changing to returning a structure more suitable for each token type, but that would be less uniform and the interface would be more complex.

Reply:
From the business's point of view, one balance is enough. Array data is not friendly to users. So, return a single struct is enough.

Reply:
Solevd, now returns a single TokenInfo structure.